### DISCUSSION TITLE
How to auto purge old recordings

### DISCUSSION THREAD

**Discussion by Sarelseekat** [blakeblackshear/frigate#1612] [Ask A Question]
Hello

I set my default retention time to 28 days. Not quite knowing how long it will take to fill my drive. My drives are now full and I need to delete recordings. Any suggestions on how to go about doing this?

Also is there a way to set Frigate to only record until the drive is lets say 80% and then auto purging old recordings?

**Comment by blakeblackshear** [comment]
Lower the retention and frigate should clear them up. You should be able to safely delete some from disk manually so frigate will start if your disk is too full.

Retention based on available disk space is not a feature yet.

**Reply by Sarelseekat** [reply]
> Lower the retention and frigate should clear them up. You should be able to safely delete some from disk manually so frigate will start if your disk is too full.
> 
> Retention based on available disk space is not a feature yet.

Thanks, Blake - for some reason it didn't clear all the recordings. Some camera's that were no longer being monitored still had recordings and those recordings stayed regardless of changing the retention setting. It did however clear enough space for things to get working again. In the end, I expanded the VM's drive capacity which also helped a lot. Hopefully, there will be a feature at some point to manage recordings directly in Frigate and maybe also select something like a Google drive location to archive old recordings. Fantastic work so far Blake.

**Comment by mkalenuik** [comment]
I am too having this issue, I am using HA OS hypervisor.  My Raid is completely full changing the retention from 10 days to 3 days and restarting Frigate did not help.  Im not sure what to do to free up space.  I'm a bit of HA noob so any help as laymen as possible would be great.

**Reply by mkalenuik** [reply]
ok, Update: So I managed to get into the server via SSH and navigated to .media/frigate to access all the clips and deleted several which allowed me to get Frigate up and running (would not run while HDD full.) Having an option to delete certain clips with in the GUI would be a great idea, also as mention above "store clips based on allocated disk space" I feel would be a popular addition. Another issue I noticed is you can not access your files via Samba when the HDD is full.
